cmd/flash/src/lib.rs

```diff
@@ -363,28 +373,102 @@ fn flashcmd(
     //
     if let Err(err) = core.load(ihex_path) {
         core.run()?;
-        Err(err)
-    } else {
-        //
-        // On Gimlet Rev B, the BOOT0 pin is unstrapped -- and during a flash,
-        // it seems to float high enough to bounce the part onto the wrong
-        // image (that is, the BOOT1 image -- which by default is the ST
-        // bootloader).  This seems to only be true when resetting immediately
-        // after flashing the part:  if there is a delay on the order of ~35
-        // milliseconds or more, the BOOT0 pin is seen as low when the part
-        // resets.  Because this delay is (more or less) harmless, we do it on
-        // all platforms, and further make it tunable.
-        //
-        let delay = subargs.reset_delay;
+        return Err(err);
+    }
+
+    //
+    // On Gimlet Rev B, the BOOT0 pin is unstrapped -- and during a flash,
+    // it seems to float high enough to bounce the part onto the wrong
+    // image (that is, the BOOT1 image -- which by default is the ST
+    // bootloader).  This seems to only be true when resetting immediately
+    // after flashing the part:  if there is a delay on the order of ~35
+    // milliseconds or more, the BOOT0 pin is seen as low when the part
+    // resets.  Because this delay is (more or less) harmless, we do it on
+    // all platforms, and further make it tunable.
+    //
+    let delay = subargs.reset_delay;
+
+    if delay != 0 {
+        std::thread::sleep(std::time::Duration::from_millis(delay));
+    }
+
+    core.reset()?;
+
+    try_program_auxflash(hubris, core)?;
```

Comment:
Worth adding a comment here explaining that it is deliberate to do this *after* resetting?

Reply:
Yup, done